### PR TITLE
feat(web-app): url strategy redefinition

### LINKED ISSUE #594: URL strategy redefinition
Should /assets/carbon-charts/library-assets turn into:
    - /libraries/carbon-charts/assets
    - /assets/libraries/carbon-charts/assets
    - something else

is a library an asset? if not, can we remove the first '/asset' before getting to the library?

Is /assets/button ever a valid url? Or is it always /assets/some-library-name/button?

At the moment, there's a mismatch between the breadcrumb and the URL (see screenshot). Assets, Libraries, and a specific library are referenced as navigation contexts on the page, but the URL is missing the indication of the specific library being nested under the set of all libraries.
image

(This applies to urls throughout the whole app, not just the library asset detail page)

see discussion on: 
https://github.com/carbon-design-system/carbon-platform/pull/509#discussion_r849667790

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #594, #723

#### Changelog

**New**

- Asset Detail Overview tab: /libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]

**Changed**

- / assets -> /
- /assets/libraries -> /libraries - library detail page
- /assets/[host]/[org]/[repo]/[library]/[ref] -> /libraries/[host]/[org]/[repo]/[library]/[ref]
- /assets/[host]/[org]/[repo]/[library]/[ref]/library-assets ->  /libraries/[host]/[org]/[repo]/[library]/[ref]/assets
- /assets/[host]/[org]/[repo]/[library]/[ref]/[asset] -> /libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset] 
- /assets/[host]/[org]/[repo]/[library]/[ref]/[asset]/usage -> /libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/usage
- /assets/[host]/[org]/[repo]/[library]/[ref]/[asset]/style -> /libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/style
- /assets/[host]/[org]/[repo]/[library]/[ref]/[asset]/code -> /libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/code 
- /assets/[host]/[org]/[repo]/[library]/[ref]/[asset]/accessibility -> /libraries/[host]/[org]/[repo]/[library]/[ref]/assets/[asset]/accessibility
- /assets/[host]/[org]/[repo]/[library]/[ref]/design-kits -> /libraries/[host]/[org]/[repo]/[library]/[ref]/design-kits
- /assets/[host]/[org]/[repo]/[library]/[ref]/versions -> /libraries/[host]/[org]/[repo]/[library]/[ref]/versions
- /assets/components -> /catalogs/components
- /assets/functions -> /catalogs/functions
- /assets/patterns -> /catalogs/patterns
- /assets/templates -> /catalogs/templates
- /assets/data-visualization -> /collections/data-visualization
- Modified redirects (next.config.mjs)
- assets.stories.js -> homepage.stories.js

**Removed**

- Rewrites

#### Testing / reviewing

Verify all new page locations match description and test navigation in general (+ internal links) to make sure nothing broke/got left behind

### CODE REVIEW

File: services/web-app/next.config.mjs [hunk 1]
```diff
@@ -80,43 +80,28 @@ const nextConfig = withMDX(
...
-          destination: '/assets',
-          permanent: false
-        },
-        {
-          source: '/assets/:host/:org/:repo/:library',
-          destination: '/assets/:host/:org/:repo/:library/latest',
-          permanent: false
-        }
-      ]
-    },
```
Comment (francinelucca) [inline services/web-app/next.config.mjs:97]: @mattrosno  I removed this, it was messing with the asset detail (turning /libraries//carbon-react/latest/assets/accordion into /libraries//carbon-react/latest/accordion/latest)
Do we need this?

**Review by alisonjoseph — APPROVED** [review]
Clicking through and everything looks like it is linking correctly! Looks like there are some lint errors on the mdx pages, I think just running lint:fix will correct. 🥳

File: services/web-app/next.config.mjs (hunk 1 above)
Comment (mattrosno) [reply services/web-app/next.config.mjs:97]: Everything in your PR looks to be working for me.

**Review by mattrosno — APPROVED** [review]
LGTM! Once we're rendering dynamic library pages (from library `navData`), let's look at the Next.js rewrites to make sure that's still working when we have `/libraries/the-library/pages/the-page`.

**Pull request merged**